File: src/CoolBytes.Core/Models/BlogPost.cs
using CoolBytes.Core.Collections;
using System;

namespace CoolBytes.Core.Models
{
    public class BlogPost
    {
        public int Id { get; private set; }
        public DateTime Date { get; internal set; }
        public Author Author { get; private set; }
        public int AuthorId { get; private set; }
        public Image Image { get; private set; }
        public int? ImageId { get; private set; }
        public UpdatableCollection<BlogPostTag> Tags { get; private set; }
        public BlogPostContent Content { get; private set; }
        public UpdatableCollection<ExternalLink> ExternalLinks { get; private set; }

        private BlogPost()
        {
            Tags = new BlogPostTagCollection();
            ExternalLinks = new ExternalLinkCollection();
        }

        public BlogPost(BlogPostContent content, Author author)
        {
            Date = DateTime.Now;
            Content = content ?? throw new ArgumentNullException(nameof(content));
            Author = author ?? throw new ArgumentNullException(nameof(author));
            Tags = new BlogPostTagCollection();
            ExternalLinks = new ExternalLinkCollection();
        }

        public void SetImage(Image image)
        {
            Image = image ?? throw new ArgumentNullException(nameof(image));
        }
    }
}

File: src/CoolBytes Website/src/app/services/blog-post.ts
import { ExternalLink } from './external-link';
import { BlogPostLink } from './blog-post-link';
import { Author } from './author';
import { BlogPostTag } from './blog-post-tag';
import { Image } from "./image";

export class BlogPost {
    id: number;
    date: Date;
    updated: Date;
    subject: string;
    contentIntro: string;
    content: string;
    tags: BlogPostTag[];
    image: Image;
    author: Author;
    links: BlogPostLink[];
    externalLinks: ExternalLink[];
}